macOS CI failing due to ffmpeg deprecation

```
 /Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:896:38: error: 'sample_fmts' is deprecated [-Werror,-Wdeprecated-declarations]
                pContext->sample_fmt = (*ppCodec)->sample_fmts ? (*ppCodec)->sample_fmts[0] : AV_SAMPLE_FMT_FLTP;
                                                   ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavcodec/codec.h:218:5: note: 'sample_fmts' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:896:64: error: 'sample_fmts' is deprecated [-Werror,-Wdeprecated-declarations]
                pContext->sample_fmt = (*ppCodec)->sample_fmts ? (*ppCodec)->sample_fmts[0] : AV_SAMPLE_FMT_FLTP;
                                                                             ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavcodec/codec.h:218:5: note: 'sample_fmts' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:897:18: error: 'supported_samplerates' is deprecated [-Werror,-Wdeprecated-declarations]
                if((*ppCodec)->supported_samplerates)
                               ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavcodec/codec.h:216:5: note: 'supported_samplerates' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:899:40: error: 'supported_samplerates' is deprecated [-Werror,-Wdeprecated-declarations]
                        pContext->sample_rate = (*ppCodec)->supported_samplerates[0];
                                                            ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavcodec/codec.h:216:5: note: 'supported_samplerates' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:900:31: error: 'supported_samplerates' is deprecated [-Werror,-Wdeprecated-declarations]
                        for(int i = 0; (*ppCodec)->supported_samplerates[i]; i++)
                                                   ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavcodec/codec.h:216:5: note: 'supported_samplerates' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
/Users/runner/work/ddnet/ddnet/src/engine/client/video.cpp:902:20: error: 'supported_samplerates' is deprecated [-Werror,-Wdeprecated-declarations]
                                if((*ppCodec)->supported_samplerates[i] == m_pSound->MixingRate())
                                               ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavcodec/codec.h:216:5: note: 'supported_samplerates' has been explicitly marked deprecated here
    attribute_deprecated
    ^
/opt/homebrew/Cellar/ffmpeg/7.1/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
#    define attribute_deprecated __attribute__((deprecated))
                                                ^
6 errors generated.
```

https://github.com/ddnet/ddnet/actions/runs/11111643515/job/30872024300?pr=9059

--- a/src/engine/client/video.cpp
+++ b/src/engine/client/video.cpp
@@ -893,13 +893,23 @@ bool CVideo::AddStream(COutputStream *pStream, AVFormatContext *pFormatContext, 
 	switch((*ppCodec)->type)
 	{
 	case AVMEDIA_TYPE_AUDIO:
-		pContext->sample_fmt = (*ppCodec)->sample_fmts ? (*ppCodec)->sample_fmts[0] : AV_SAMPLE_FMT_FLTP;
-		if((*ppCodec)->supported_samplerates)
+	{
+		const AVSampleFormat *pSampleFormats = nullptr;
+		const int *pSampleRates = nullptr;
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(61, 13, 100)
+		avcodec_get_supported_config(pContext, *ppCodec, AV_CODEC_CONFIG_SAMPLE_FORMAT, 0, (const void **)&pSampleFormats, nullptr);
+		avcodec_get_supported_config(pContext, *ppCodec, AV_CODEC_CONFIG_SAMPLE_RATE, 0, (const void **)&pSampleRates, nullptr);
+#else
+		pSampleFormats = (*ppCodec)->sample_fmts;
+		pSampleRates = (*ppCodec)->supported_samplerates;
+#endif
+		pContext->sample_fmt = pSampleFormats ? pSampleFormats[0] : AV_SAMPLE_FMT_FLTP;
+		if(pSampleRates)
 		{
-			pContext->sample_rate = (*ppCodec)->supported_samplerates[0];
-			for(int i = 0; (*ppCodec)->supported_samplerates[i]; i++)
+			pContext->sample_rate = pSampleRates[0];
+			for(int i = 0; pSampleRates[i]; i++)
 			{
-				if((*ppCodec)->supported_samplerates[i] == m_pSound->MixingRate())
+				if(pSampleRates[i] == m_pSound->MixingRate())
 				{
 					pContext->sample_rate = m_pSound->MixingRate();
 					break;
@@ -921,6 +931,7 @@ bool CVideo::AddStream(COutputStream *pStream, AVFormatContext *pFormatContext, 
 		pStream->m_pStream->time_base.num = 1;
 		pStream->m_pStream->time_base.den = pContext->sample_rate;
 		break;
+	}
 
 	case AVMEDIA_TYPE_VIDEO:
 		pContext->codec_id = CodecId;

